kratos/tests/cpp_tests/geometries/test_geometry_id.cpp

```diff
@@ -0,0 +1,50 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//                   Pooyan Dadvand
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "geometries/quadrilateral_2d_4.h"
+
+#include "tests/cpp_tests/geometries/test_geometry.h"
+
+#include "containers/pointer_vector.h"
+
+namespace Kratos {
+namespace Testing {
+
+    Quadrilateral2D4<Point> GenerateQuadrilateralWithIdPointer(IndexType GeometryId)
+    {
+        Quadrilateral2D4<Point> quad(
+            Point::Pointer(new Point(0.0, 0.0, 0.0)),
+            Point::Pointer(new Point(1.0, 0.0, 0.0)),
+            Point::Pointer(new Point(1.0, 1.0, 0.0)),
+            Point::Pointer(new Point(0.0, 1.0, 0.0))
+                );
+        quad.SetId(GeometryId);
+        return quad;
+    }
+
+    ///// Test Geometry Id
+    KRATOS_TEST_CASE_IN_SUITE(TestGeometryId, KratosCoreGeometriesFastSuite) {
+        auto quad = GenerateQuadrilateralWithIdPointer(1);
+
+        //// Check general information, input to ouput
+        KRATOS_CHECK_EQUAL(quad.Id(), 1);
+    }
+} // namespace Testing.
```

Comment:
Very few tests. I would add tests for the flags to be sure they are working as they should

Reply:
You may add these tests to the test_geometries to have them all together.

Reply:
I added a set of tests to the geometry_container, which include the use of the id. If you agree I would remove this test here completely.

Reply:
I would add specific tests for id and checks if they are self-assigned etc and put all of them including this one in the test_geometries.cpp. 

One of the geometry_containers should check only the operations of the container. (of course, they will fail if id is not correct but we need tests that check this separately)